.github/workflows/wheels.yml

```diff
@@ -1,49 +1,63 @@
-# This file is autogenerated by maturin v1.7.4
-# To update, run
-#
-#    maturin generate-ci github
-#
-name: wheels
+name: Build wheels
 
 on:
   push:
     tags:
-      - "py-v*"
+      - "*"
+  pull_request:
+    branches: [main]
   workflow_dispatch:
 
 permissions:
   contents: read
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   linux:
     runs-on: ${{ matrix.platform.runner }}
     strategy:
+      fail-fast: false
+
       matrix:
         platform:
           - runner: ubuntu-latest
             target: x86_64
-          - runner: ubuntu-latest
-            target: x86
+            manylinux: auto
+          # TODO: gcc can't find a suitable -lgcc on 32-bit linux
+          # - runner: ubuntu-latest
+          #   target: x86
+          #   manylinux: auto
           - runner: ubuntu-latest
             target: aarch64
+            manylinux: auto
           - runner: ubuntu-latest
             target: armv7
+            manylinux: auto
           - runner: ubuntu-latest
             target: s390x
+            manylinux: auto
           - runner: ubuntu-latest
             target: ppc64le
+            manylinux: auto
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
-        with:
-          python-version: 3.x
+
       - name: Build wheels
         uses: PyO3/maturin-action@v1
         with:
           target: ${{ matrix.platform.target }}
-          args: --release --out dist --find-interpreter
+          args: --release --out dist -i 3.10 -i 3.11 -i 3.12 -i 3.13
```

Comment:
Can build just one wheel per platform using abi3? See https://www.maturin.rs/bindings.html?highlight=abi3#py_limited_apiabi3 (set `pyo3 = { version = "0.23.3", features = ["abi3-py310"] }` in your Cargo.toml)

Reply:
that was something I copied from `arro3`'s setup, so no personal preferences (and as I just started out with compiled extensions I don't really have the knowledge on what to prefer here)